components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/claims/impl/DefaultClaimHandler.java

```diff
@@ -1100,6 +1107,112 @@ private void handleRoleClaim(AuthenticationContext context, Map<String, String>
         }
     }
 
+    /**
+     * Add the application roles of federated user to remote claims.
+     *
+     * @param stepConfig       StepConfig of current step.
+     * @param context          AuthenticationContext of current authentication flow.
+     * @param idPClaimMappings Claim mappings of IdP of the current step.
+     * @param remoteClaims     Remote claims of the current step authenticated user.
+     * @throws FrameworkException Exception on handling application roles for federated user.
+     */
+    protected void handleApplicationRolesForFederatedUser(StepConfig stepConfig, AuthenticationContext context,
+                                                        ClaimMapping[] idPClaimMappings,
+                                                        Map<String, String> remoteClaims)
+            throws FrameworkException {
+
+        // IdP claim mappings should be available and the current step should be a subject attribute step.
+        if (idPClaimMappings == null || !stepConfig.isSubjectAttributeStep()) {
+            return;
+        }
+        // Get the remote claim URI of the application role claim.
+        String idpAppRoleClaimUri = Arrays.stream(idPClaimMappings)
```

Comment:
```suggestion
        String remoteClaimURIOfAppRoleClaim = Arrays.stream(idPClaimMappings)
```

Reply:
Addressed in 734a7bff58df0ecbf63400a8d7f3f61cf3ff22b4